### PR TITLE
[API] Checkout addressing step added

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Command/OrderTokenValueAwareInterface.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Command;
+
+interface OrderTokenValueAwareInterface
+{
+    public function getOrderTokenValue(): ?string;
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/Command/OrderTokenValueAwareInterface.php:18]:
```suggestion
    public function getOrderTokenValue(): ?string;
    
```
?

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * @When I specify the email as :email
+     */
+    public function iSpecifyTheEmailAs(string $email): void
+    {
+        $this->content['email'] = $email;
+    }
+
+
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:65]: Unnecessary blank line

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    /**
+     * @When I specify the email as :email
+     */
+    public function iSpecifyTheEmailAs(string $email): void
+    {
+        $this->content['email'] = $email;
+    }
+
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:64]:
```suggestion
```

File: src/Sylius/Bundle/ApiBundle/Command/Checkout/AddressOrder.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    public $orderTokenValue;
+
+    /**
+     * @var string
+     * @psalm-immutable
+     */
+    public $email;
+
+    /**
+     * @var Address
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Command/Checkout/AddressOrder.php:32]:
```suggestion
     * @var AddressInterface
```

File: src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Checkout/AddressOrderHandlerSpec.php [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    function let(
+        OrderRepositoryInterface $orderRepository,
+        FactoryInterface $customerFactory,
+        ObjectManager $manager,
+        StateMachineFactoryInterface $stateMachineFactory
+    ): void {
+        $this->beConstructedWith($orderRepository, $customerFactory, $manager, $stateMachineFactory);
+    }
+
+    function it_handles_order_shipment_addressing(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Checkout/AddressOrderHandlerSpec.php:39]:
```suggestion
    function it_handles_addressing_an_order(
```

**Pull request merged**